source/material_model/visco_plastic.cc

```diff
@@ -191,15 +197,34 @@ namespace aspect
           // Convert friction angle from degrees to radians
           double phi = angles_internal_friction[j] * numbers::PI/180.0;
 
+          // Assing cohesions to a new variable
+          double coh = cohesions[j];
+
+          // Strain weakening
+          double strain_ii = 0;
+          if (use_strain_weakening == true)
+
```

Comment:
Ah, it might work like this as well, but I would prefer if you would do everything up to line 220 only if `use_strain_weakening` is set to true. This way it is clearer that all of that only matters for strain weakening. And the declaration of `strain_ii` might go into that if-loop as well if you do not need it later on.